ocs_ci/ocs/resources/pod.py

```diff
@@ -695,6 +701,107 @@ def get_ocs_operator_pod(ocs_label=constants.OCS_OPERATOR_LABEL, namespace=None)
+    label=constants.MANAGED_ALERTMANAGER_LABEL, namespace=None
+):
+    """
+    Get alertmanager-managed-ocs-alertmanager pods in the cluster
+
+    Args:
+        label (str): Label associated with alertmanager-managed-ocs-alertmanager pods
+        namespace (str): Namespace in which alertmanager-managed-ocs-alertmanager pods are residing
+
+    Returns:
+        list: Pod objects of alertmanager-managed-ocs-alertmanager pods
+
+    """
+    namespace = namespace or config.ENV_DATA["cluster_namespace"]
+    alertmanager_managed_pods = get_pods_having_label(label, namespace)
+    return [
+        Pod(**alertmanager_managed)
+        for alertmanager_managed in alertmanager_managed_pods
+    ]
+
+
+def get_ocs_osd_controller_manager_pod(
+    label=constants.MANAGED_CONTROLLER_LABEL, namespace=None
+):
+    """
+    Get ocs-osd-controller-manager pod in the cluster
+
+    Args:
+        label (str): Label associated with ocs-osd-controller-manager pod
+        namespace (str): Namespace in which ocs-osd-controller-manager pod is residing
+
+    Returns:
+        Pod: Pod object of ocs-osd-controller-manager pod
+
+    """
+    namespace = namespace or config.ENV_DATA["cluster_namespace"]
+    # odf-operator-controller-manager pod also have the same label. Select ocs-osd-controller-manager pod only.
+    controller_manager_pods = [
+        controller_manager
+        for controller_manager in get_pods_having_label(label, namespace)
+        if "ocs-osd-controller-manager" in controller_manager["metadata"]["name"]
+    ]
+    return Pod(**controller_manager_pods[0])
+
+
+def get_prometheus_managed_ocs_prometheus_pod(
+    label=constants.MANAGED_PROMETHEUS_LABEL, namespace=None
+):
+    """
+    Get prometheus-managed-ocs-prometheus pod in the cluster
+
+    Args:
+        label (str): Label associated with prometheus-managed-ocs-prometheus pod
+        namespace (str): Namespace in which prometheus-managed-ocs-prometheus pod is residing
+
+    Returns:
+        Pod: Pod object of prometheus-managed-ocs-prometheus pod
+
+    """
+    namespace = namespace or config.ENV_DATA["cluster_namespace"]
+    prometheus_managed_ocs_prometheus = get_pods_having_label(label, namespace)
+    return Pod(**prometheus_managed_ocs_prometheus[0])
+
+
+def get_prometheus_operator_pod(
+    label=constants.PROMETHEUS_OPERATOR_LABEL, namespace=None
+):
+    """
+    Get prometheus-operator pod in the cluster
+
+    Args:
+        label (str): Label associated with prometheus-operator pod
+        namespace (str): Namespace in which prometheus-operator pod is residing
+
+    Returns:
+        Pod: Pod object of prometheus-operator pod
+
+    """
+    namespace = namespace or config.ENV_DATA["cluster_namespace"]
+    prometheus_operator = get_pods_having_label(label, namespace)
+    return Pod(**prometheus_operator[0])
+
+
+def get_ocs_provider_server_pod(label=constants.PROVIDER_SERVER_LABEL, namespace=None):
+    """
```

Comment:
Do we really need so many functions that are so similar? Can't we have a more general one that takes label as an argument?

Reply:
@ebondare 
The function 'get_pods_having_label' which is used inside this function, is actually the generic one. The idea was to keep the conditional statements in the Disruptions class in the same way it is originally designed. If separate functions are not present, the function 'set_resource' in the class Disruptions will look bulky.
Still it can work without the separate functions. We can change the original structure(which is not part of this PR) of the function  'set_resource' if needed. WDYT ?